### PR TITLE
wrench: Cleanup unstride()

### PR DESCRIPTION
**Pull request by jrmuizel**
This switches to using an iterator and factors out the saving.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/689)

### CODE REVIEW

File: wrench/src/yaml_frame_writer.rs [hunk 1]
```diff
@@ -368,6 +342,17 @@ impl YamlFrameWriter {
             }
         };
 
+        if data.stride == data.width * bpp {
+            save_buffer(&path_file, &bytes, data.width, data.height, ColorType::RGB(8)).unwrap();
+        } else {
+            // takes a buffer with a stride and copies it into a new buffer that has stride == width
+            let tmp : Vec<_>  = bytes[..].chunks(data.stride as usize)
```
Comment (kvark) [inline wrench/src/yaml_frame_writer.rs]: nit: extra spaces here and misaligned following iterator modifiers

File: wrench/src/yaml_frame_writer.rs [hunk 2]
```diff
@@ -368,6 +342,17 @@ impl YamlFrameWriter {
             }
         };
 
+        if data.stride == data.width * bpp {
+            save_buffer(&path_file, &bytes, data.width, data.height, ColorType::RGB(8)).unwrap();
+        } else {
+            // takes a buffer with a stride and copies it into a new buffer that has stride == width
```
Comment (kvark) [inline wrench/src/yaml_frame_writer.rs]: we might want to `assert!(data.stride > data.width * bpp)` here

**Comment by jrmuizel** [comment]
I've addressed the review comments.

**Comment by kvark** [comment]
Thanks!
@bors-servo r+

**Pull request merged**